djangocms_versioning_filer/helpers.py

```diff
@@ -74,3 +76,31 @@ def check_file_exists_in_folder(file_obj):
     return check_file_label_exists_in_folder(
         file_obj.label, file_obj.folder, exclude_file_pks=exclude_file_pks,
     )
+
+
+def filename_exists(request, folder_id=None):
+    from filer.models import Folder, File
+
+    FILE_EXISTS = _('File name already exists')
+
+    try:
+        # Get folder
+        folder = Folder.objects.get(pk=folder_id)
+    except Folder.DoesNotExist:
+        # if folder not exists then not proceeding further check and return
+        return
+
+    if folder:
+        if len(request.FILES) == 1:
+            # dont check if request is ajax or not, just grab the file
```

Comment:
Why are we not checking if it is Ajax? Just trying to understand the rationale behind having a comment explaining what is not there?